SpatialGDK/Source/SpatialGDKTests/Private/LocalDeploymentManager/LocalDeploymentManagerTest.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "LocalDeploymentManager.h"
+#include "SpatialGDKDefaultLaunchConfigGenerator.h"
+#include "SpatialGDKDefaultWorkerJsonGenerator.h"
+#include "SpatialGDKEditorSettings.h"
+
+#include "CoreMinimal.h"
+
+#define LOCALDEPLOYMENT_TEST(TestName) \
+	TEST(Services, LocalDeployment, TestName)
+
+namespace
+{
+	// TODO: UNR-1969 - Prepare LocalDeployment in CI pipeline
+	static const double MAX_WAIT_TIME_FOR_LOCAL_DEPLOYMENT_OPERATION = 10.0;
+
+	// TODO: UNR-1964 - Move EDeploymentState enum to LocalDeploymentManager
+	enum class EDeploymentState { IsRunning, IsNotRunning };
+
+	static const FName AutomationWorkerType = TEXT("AutomationWorker");
+	static const FString AutomationLaunchConfig = TEXT("Improbable/AutomationLaunchConfig.json");
+
+	FLocalDeploymentManager* GetLocalDeploymentManager()
+	{
+		FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+		FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+		return LocalDeploymentManager;
+	}
+
+	FSpatialLaunchConfigDescription GenerateLaunchConfigDescription()
+	{
+		FSpatialLaunchConfigDescription LaunchConfigDescription;
+		FWorkerTypeLaunchSection UnrealWorkerDefaultSetting;
+		UnrealWorkerDefaultSetting.WorkerTypeName = FName(AutomationWorkerType);
+		UnrealWorkerDefaultSetting.Rows = 1;
+		UnrealWorkerDefaultSetting.Columns = 1;
+		UnrealWorkerDefaultSetting.bManualWorkerConnectionOnly = true;
+		LaunchConfigDescription.ServerWorkers.Reset();
+		LaunchConfigDescription.ServerWorkers.Add(UnrealWorkerDefaultSetting);
+
+		return LaunchConfigDescription;
+	}
+
+	bool GenerateWorkerPB()
+	{
+		FString BuildConfigArgs = TEXT("worker build build-config");
+		FString WorkerBuildConfigResult;
+		int32 ExitCode;
+		const FString SpatialExe(TEXT("spatial.exe"));
```

Comment:
nit: This variable is used in a few places now, would be good to centralise.

Reply:
This is something that this PR https://github.com/spatialos/UnrealGDK/pull/1303 will do.